### PR TITLE
split out the key based into a sperate component

### PR DESCRIPTION
**Pull request by edewit**
it's made it's base component to hard to reuse and was only used once

### CODE REVIEW

File: src/clients/authorization/KeyBasedAttributeInput.tsx [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+              aria-label={t("roles:addAttributeText")}
+              id="plus-icon"
+              variant="link"
+              className="kc-attributes__plus-icon"
+              onClick={() => {
+                append({ key: "", value: "" });
+                setIsKeyOpenArray([...isKeyOpenArray, false]);
+              }}
+              icon={<PlusCircleIcon />}
+              isDisabled={!watchLastValue}
```
Comment (jenny-s51) [inline src/clients/authorization/KeyBasedAttributeInput.tsx:258]: Currently this is always disabled even if a value is typed into the attribute value input field.

File: src/clients/authorization/KeyBasedAttributeInput.tsx [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+                      >
+                        {attribute.name}
+                      </SelectOption>
+                    ))}
+                  </Select>
+                )}
+              />
+            </Td>
+            <ValueInput
+              attribute={attribute}
```
Comment (jenny-s51) [inline src/clients/authorization/KeyBasedAttributeInput.tsx:229]:
```suggestion
              name={name}
              attribute={attribute}
```

**Review by jenny-s51 — CHANGES_REQUESTED** [review]
Good idea splitting this up into a separate component @edewit !

I tested this in the Authorization -> Evaluate tab of Clients and noticed that whatever was typed into the "value" field wasn't being sent with the data. 

Passing the `name` prop to `ValueInput` seems to fix it!

File: src/clients/authorization/KeyBasedAttributeInput.tsx (hunk 2 above)
Comment (edewit) [reply src/clients/authorization/KeyBasedAttributeInput.tsx:229]: good catch...

**Review by jonkoops — CHANGES_REQUESTED** [review]

**Review by jonkoops — CHANGES_REQUESTED** [review]
Changes look good, but the tests are failing. @edewit can you take a look?

**Review by jenny-s51 — APPROVED** [review]

**Review by jonkoops — APPROVED** [review]

**Pull request merged**